edx_exams/apps/core/api.py

```diff
@@ -158,6 +158,25 @@ def get_exam_attempt_time_remaining(exam_attempt, now=None):
     return secs_to_end
 
 
+def check_if_exam_timed_out(exam_attempt):
+    """
+    If an exam attempt's time remaining = 0 AND is in-progress/not submitted,
+    change the status of an exam attempt to 'submitted', then return attempt_id.
+    Return None otherwise.
+    """
+    IN_PROGRESS_STATUSES = [
+        ExamAttemptStatus.started,
+        ExamAttemptStatus.ready_to_submit,
+    ]
+    if exam_attempt.status in IN_PROGRESS_STATUSES and get_exam_attempt_time_remaining(exam_attempt) == 0:
+        updated_attempt_id = update_attempt_status(exam_attempt.id, ExamAttemptStatus.submitted)
```

Comment:
Varsha's comment about logging made me think that we will want to log when an attempt is auto submitted because it's timed out. Adding that to this function would make sense 👍